modules/collabora_online_group/tests/src/Kernel/PermissionTest.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\collabora_online_group\Kernel;
+
+use Drupal\Tests\collabora_online_group\Traits\GroupRelationTrait;
+use Drupal\Tests\group\Kernel\GroupKernelTestBase;
+use Drupal\Tests\media\Traits\MediaTypeCreationTrait;
+
+/**
+ * Tests group permissions.
+ */
+class PermissionTest extends GroupKernelTestBase {
+
+    use MediaTypeCreationTrait;
+    use GroupRelationTrait;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected static $modules = [
+        'file',
+        'media',
+        'group',
+        'groupmedia',
+    ];
+
+    /**
+     * Tests that group permissions are properly created.
+     */
+    public function testGroupPermissions(): void {
+        // Generate types: groups and medias.
+        $group_type_1 = $this->createGroupType(['id' => 'group_1']);
+        $group_type_2 = $this->createGroupType(['id' => 'group_2']);
+        $group_type_3 = $this->createGroupType(['id' => 'group_3']);
+        $this->createMediaType('file', ['id' => 'media_1']);
+        $this->createMediaType('file', ['id' => 'media_2']);
```

Comment:
We could use arrays instead of the individual variables.
```php
$group_types = [
  'group_1' => $this->createGroupType(['id' => 'group_1']),
  'group_2' => $this->createGroupType(['id' => 'group_2']),
  'group_3' => $this->createGroupType(['id' => 'group_3']),
];
```
Also, maybe they should be called 'group_type_1' or 'type_1' instead of 'group_1'.
My first thought when seeing 'group_1' is that this is a group, not a group type.

Alternatively we could use names like 'sports club', 'interest group', etc, but I don't think it would add much value here.

Reply:
Actually, we don't need to set ID's for group types, we only use media types so I remove this.
For media types changed ID's.